### PR TITLE
Simplify signature of handler install

### LINKED ISSUE #914: Simplify signature of handler install
We keep adding params to the signature of `HandlerRegistry::install` to handle new behaviour for specific RPCs - this is mostly my fault. I think we should instead have a basic install function (`install(name, func, ReadWrite)`, and set other fields explicitly. A sensible way to do this is for `install` to return an iterator/handle that we can set additional fields on, similar to `CLI11` options, so we can write `handler->execute_locally = true; handler->schema = ...;`.

### PR DESCRIPTION
**Pull request by eddyashton**
Resolves #914.

We have several parameters to tweak the behaviour of installed RPC handlers, and we're likely to add more. Rather than needing to specify all of these in the call to `install` (with signficant overload complications), they should be set manually on the few handlers they actually affect.

This PR lets you set those parameters in 2 ways. Manually:

```
auto& foo_handler = install("FOO", foo_func, Read);
foo_handler.execute_locally = true;
foo_handler.params_schema = ...;
```

Or with chaining methods:

```
install("FOO", foo_func, Read)
  .set_execute_locally(true)
  .set_params_schema(...);
```

I prefer the second, but I think its worth keeping both in general.

### CODE REVIEW

**Comment by cimetrics** [comment]
simple_install_signature@[6057 aka 20200317.10](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=6057&view=results) vs master ewma over 30 builds from [5763](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=5763&view=results) to [6053](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=6053&view=results)
images

File: src/node/rpc/handlerregistry.h [hunk 1]
```diff
@@ -68,80 +129,16 @@ namespace ccf
...
-        rw,
-        params_schema,
-        result_schema,
-        require_client_signature,
-        execute_locally);
-    }
-
-    template <typename T, typename... Ts>
-    void install_with_auto_schema(const std::string& method, Ts&&... ts)
+    Handler& install(
```
Comment (jumaffre) [inline src/node/rpc/handlerregistry.h]: Might be worth documenting the new `@return` here?

File: src/apps/logging/logging.cpp [hunk 2]
```diff
@@ -195,24 +195,19 @@ namespace ccfapp
       };
       // SNIPPET_END: log_record_prefix_cert
 
-      install_with_auto_schema<LoggingRecord::In, bool>(
-        Procs::LOG_RECORD, json_adapter(record), Write);
+      install(Procs::LOG_RECORD, json_adapter(record), Write)
+        .set_auto_schema<LoggingRecord::In, bool>();
       // SNIPPET: install_get
```
Comment (jumaffre) [inline src/apps/logging/logging.cpp]: Outside of this PR: I think we need to make this a `SNIPPET_START`, `SNIPPET_END` as it only renders the first line:
image

File: src/node/rpc/handlerregistry.h (hunk 1 above)
Comment (eddyashton) [reply src/node/rpc/handlerregistry.h]: Added.

File: src/apps/logging/logging.cpp (hunk 2 above)
Comment (eddyashton) [reply src/apps/logging/logging.cpp]: Added in this PR, since I'm touching it anyway.

**Comment by achamayou** [comment]
@eddyashton this is great! I think we should have the accessors only, and make sure they're exposed in doxygen/sphinx.

**Pull request merged**